host/xtest/sdp_basic.c

```diff
@@ -0,0 +1,521 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <err.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <tee_client_api.h>
+#include <tee_client_api_extensions.h>
+#include <unistd.h>
+
+#include "include/uapi/linux/ion.h"
+#include "ta_sdp_basic.h"
+#include "crypto_common.h"
+
+/*
+ * SDP basic test setup overview.
+ *
+ * - A dedicated trusted application (SDP basic TA) supports 3 commands:
+ *   - 'inject' data from a nonsecure buffer into a secure buffer
+ *   - 'transform' data inside a secure buffer (bitwise invert + unsigned incr)
+ *   - 'dump' data from a secure buffer into a nonsecure buffer
+
+ * - This test client application (CA) invokes the TA for these 3 operations,
+ *   inject random value, trasforming them then dump them.
+ *
+ * To do so, CA allocates a 'SDP secure buffer' and invoke the TA for these 3
+ * operations (inject then transform then dump) over the allocate buffer.
+ *
+ * The secure buffer is currently allocation through ION support adn
+ * registered to OP-TEE and as shared memory.
+ *
+ * To enhance test coverage against buffer alignement usecase, the CA invokes
+ * the TA with a variable offset inside the buffer. As CA injects random data
+ * into the buffer, the CA uses one of the random bytes to set the value of the
+ * offset in the accessed secure buffer.
+ *
+ * For debugging support, the CA may map (in nonsecure world) the secure
+ * buffer to read its content. As this is unsafe on a hardened platform, this
+ * operation is default disable. When enable, error only print out a warning
+ * trace but does not actually fail the test. This also give an easy way to
+ * check that some HW complains on access violation when nonsecure accesses
+ * secure data.
+ */
+
+static int verbosity = 1;
+
+struct tee_ctx {
+	TEEC_Context ctx;
+	TEEC_Session sess;
+};
+
+/* exported to xtest */
+int allocate_ion_buffer(size_t size, int heap_id);
+
+/* non zero value forces buffer to be mappeable from nonsecure */
+#define BUF_MUST_MAP	0
+
+#define DEFAULT_ION_HEAP_TYPE	ION_HEAP_TYPE_UNMAPPED
+
+/*
+ * Warm when nonsecure maps SDP buffer and found unexpected content.
+ * Since mapping SDP buffers from nonsecure is unsafe, such accesses
+ * from nonsecure shall not trigger errors, only warnings.
+ */
+static int warm_ns_access = 1;
```

Comment:
s/Warm/Warn/
a/warm/warn/

Is this variable actually used somewhere? I can't find it.